### PR TITLE
provide the flag labware admin page and the update the labware details page to dislay the recorded flags

### PR DESCRIPTION
**Pull request by sabrine33**

### CODE REVIEW

File: src/pages/FlagLabware.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  const checkRelatedFlagges = useCallback(
+    async (labwares: LabwareFieldsFragment[], foundLabware: LabwareFieldsFragment): Promise<string[]> => {
+      return stanCore
+        .GetLabwareFlagDetails({
+          barcodes: [foundLabware.barcode]
+        })
+        .then((response) => {
+          setRelatedFlags(response.labwareFlagDetails);
+          return [];
+        });
```
Comment (seenanair) [inline src/pages/FlagLabware.tsx:81]: Using await will improve readability, also allows  try/catch blocks for error handling

File: src/pages/FlagLabware.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    },
+    [stanCore]
```
Comment (seenanair) [inline src/pages/FlagLabware.tsx:83]: Would it not require `setRelatedFlags` in dependency array?

File: src/pages/FlagLabware.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+                              {relatedFlags.length > 0 && <LabwareFlagDetails flagDetails={relatedFlags} />}
+
+                              <div className="mt-6">
+                                <Heading level={3}>Description</Heading>
+                                <MutedText>Please enter a reason for flagging the labware.</MutedText>
+
+                                <Field
+                                  as={'textarea'}
+                                  placeholder="Enter your the reason for flagging the labware here."
```
Comment (seenanair) [inline src/pages/FlagLabware.tsx:135]: Is it required to have both MutedText and placeholder with the same information?

File: src/pages/FlagLabware.tsx (hunk 2 above)
Comment (sabrine33) [reply src/pages/FlagLabware.tsx:83]: React guarantees that setState function identity is stable and won't change on re-renders. This is why it's safe to omit from the useEffect or useCallback dependency list.

File: src/pages/FlagLabware.tsx (hunk 3 above)
Comment (sabrine33) [reply src/pages/FlagLabware.tsx:135]: not required, I will remove the placeholder and keep the muted text, so it is coherent with the other pages

**Review by seenanair — APPROVED** [review]

**Comment by sabrine33** [comment]
merged to devel

**Pull request merged**